vlab_inf_common/vmware/virtual_machine.py

```diff
@@ -727,3 +730,180 @@ def _get_upload_url(vcenter, the_vm, creds, upload_path, file_size, file_attribu
+    return vm_ovf.ovfDescriptor
+
+
+class ProgressChimer(threading.Thread):
+    """Keeps the lease alive while downloading a VMDK from vSphere.
+
+    Using in a ``with`` statement ensures the lease is closed, and makes your code
+    cleaner!
+
+    :param lease: Required to export a VM to an OVA.
+    :type lease: vim.HttpNfcLease
+
+    :param log: An object for writing debug/progress messages.
+    :type log: logging.Logger
+
+    :param update_interval: How often renew/update the lease, in seconds.
+    :type update_interval: Integer
+    """
+    def __init__(self, lease, log, update_interval=10):
+        super().__init__()
+        self._lease = lease
+        self._keep_running = True
+        self._update_interval = update_interval
+        self.start()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exce_type, exec_value, exce_traceback):
+        self.complete()
+
+    def complete(self):
+        self._lease.HttpNfcLeaseProgress(100)
+        self._lease.HttpNfcLeaseComplete()
+        self._keep_running = False
+        self.join()
+
+    def run(self):
+        while self._keep_running:
+            self._lease.HttpNfcLeaseProgress(50)
+            time.sleep(self._update_interval)
+
+
+def _block_on_lease(lease):
+    """Blocks execution until the lease is usable or fails
+
+    :Returns: None
+
+    :param lease: Required to export a VM to an OVA.
+    :type lease: vim.HttpNfcLease
+    """
+    # this amounts to waiting upwards of 990 seconds, with linear backoff
+    for i in range(45):
+        if lease.state == vim.HttpNfcLease.State.ready:
+            break
+        elif lease.state == vim.HttpNfcLease.State.error:
+            raise RuntimeError("VM Export lease error: {}".format(lease.state.error))
+        else:
+            time.sleep(i)
+    else:
+        raise RuntimeError("Lease never became ready")
+
+
+def make_ova(vcenter, the_vm, template_dir, log):
+    """Export a virtual machine into an OVA. The returned string is the location
+    of the new OVA file.
+
+    :Returns: String
+
+    :param vcenter: The instantiated connection to vCenter
+    :type vcenter: vlab_inf_common.vmware.vCenter
+
+    :param the_vm: The name of the VM
+    :type the_vm: String
+
+    :param template_dir: The folder to save the new OVA to.
+    :type template_dir: String
+
+    :param log: A message for writing progress/debug messages.
+    :type log: logging.Logger
+    """
+    ova_location = ''
+    power(the_vm, 'off')
+    lease = the_vm.ExportVm()
+    _block_on_lease(lease)
+    with ProgressChimer(lease, log):
+        save_location = os.path.join(template_dir, the_vm.name)
+        os.makedirs(save_location, exists_ok=True)
+        device_ovfs = []
+        for device in lease.info.deviceUrl:
+            device_ovf = download_vmdk(save_location, vcenter.cookie(), device, log)
+            device_ovfs.extend(device_ovf)
+        vm_ovf_xml = get_vm_ovf_xml(the_vm, device_ovfs, vcenter)
```

Comment:
No need to keep the `ProgressChimer` alive. Why not de-dent this chunk and it's related lines?

Reply:
Hmm... The [ref code](https://github.com/vmware/pyvmomi-community-samples/blob/master/samples/export_vm.py#L287) keeps it alive until the very end 🤔. I don't think we need the list to get the VM OVF. Let's de-dent it now, and if it blows up, we'll undo that. Easier than trying to remember to de-dent when everything works.